test/injector/InjectorTest.ts

```diff
@@ -374,4 +374,21 @@ import {InjectionMapping} from "../../src/injector/data/InjectionMapping";
         this.injector = null;
     }
 
+    @test("Instance must be available in Injector before @PostConstruct is invoked")
+    @timeout(10)
+    isInstanceInInjectorOnPostConstruct(done:() => void) {
+        this.injector.map(CustomModelWithPostConstruct).asSingleton();
+        let instance:CustomModelWithPostConstruct;
```

Comment:
The instance can get the default value and be const instead of let

Reply:
Not in this case, it's referenced before it's set

Reply:
Correct, didnt notice